### ISSUE TITLE
Build failure: vagrant

### ISSUE THREAD

**Issue by andersk** [NixOS/nixpkgs#348938]
### Steps To Reproduce

1. build `vagrant`

### Build log

```
Running phase: installCheckPhase
Vagrant failed to initialize at a very early stage:
The plugins failed to initialize correctly. This may be due to manual
modifications made within the Vagrant home directory. Vagrant can
attempt to automatically correct this issue by running:
  vagrant plugin repair
If Vagrant was recently updated, this error may be due to incompatible
versions of dependencies. To fix this problem please remove and re-install
all plugins. Vagrant can attempt to do this automatically by running:
  vagrant plugin expunge --reinstall
Or you may want to try updating the installed plugins to their latest
versions:
  vagrant plugin update
Error message given during initialization: can't activate rexml-3.3.6, already activated rexml-3.2.5
```

### Additional context

- hashicorp/vagrant_cloud#86

### Notify maintainers

@tylerjl

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.56, NixOS, 24.11 (Vicuna), 24.11.git.0846c6c8a7db`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(anders): `""`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by roblabla** [comment]
Duplicate of https://github.com/NixOS/nixpkgs/issues/348108 ?

**Comment by andersk** [comment]
Workaround (if you don’t need libvirt support): `vagrant.override { withLibvirt = false; }` still works.

**Comment by johanneskastl** [comment]
> Workaround (if you don’t need libvirt support): `vagrant.override { withLibvirt = false; }` still works.

Sorry for the newbie question, but how would I use this override in a home-manager file?

**Comment by lucc** [comment]
@johanneskastl because there is no option for vagrant you should be able to put the package into `home.packages`:
~~~nix
home.packages = [
  (vagrant.override {withLibvirt=false;})
];
~~~

Otherwise you could check the FAQ: https://nix-community.github.io/home-manager/index.xhtml#_how_do_i_change_the_package_used_by_a_module

**Comment by johanneskastl** [comment]
Thanks, that was almost correct. This worked:

```
    home.packages = [ 
      (pkgs.vagrant.override {withLibvirt=false;})
    ];
```

**Comment by johanneskastl** [comment]
Unfortunately, the workaround allows building the vagrant derivation. But installation of the vagrant-libvirt plugins fails...

**Comment by tylerjl** [comment]
Fixed build in #362117; some additional testing on Darwin as noted in the PR would be appreciated.

**Comment by bjornfor** [comment]
I was curious to see what caused vagrant to stop building. According to git bisect it is a48704428792db4901fd41bc940f4e610a4edd00 ("ruby: 3.3.4 -> 3.3.5 (#340137)").

**Comment by bjornfor** [comment]
Fixed by 3cb2fa6e0014502d57d681d5561f2db1e2db5917 (and backported to release-24.11 in ec4dd829e5a18d4808aa5fe7d582c8ded6b544fc). Thank you @tylerjl!

**Comment by johanneskastl** [comment]
> Fixed by [3cb2fa6](https://github.com/NixOS/nixpkgs/commit/3cb2fa6e0014502d57d681d5561f2db1e2db5917) (and backported to release-24.11 in [ec4dd82](https://github.com/NixOS/nixpkgs/commit/ec4dd829e5a18d4808aa5fe7d582c8ded6b544fc)). Thank you @tylerjl!

Thanks to everyone involved @bjornfor  @tylerjl, very nice to see that fixed.
